src/display/editor/annotation_editor_layer.js

```diff
@@ -589,6 +636,96 @@ class AnnotationEditorLayer {
     this.#uiManager.unselect(editor);
   }
 
+  /**
+   * SelectionChange callback.
+   * @param {Event} _event
+   */
+  selectionStart(_event) {
+    this.#textLayer?.div.addEventListener(
+      "pointerup",
+      this.#boundPointerUpAfterSelection,
+      { once: true }
+    );
+  }
+
+  /**
+   * Called when the user releases the mouse button after having selected
+   * some text.
+   * @param {PointerEvent} event
+   */
+  pointerUpAfterSelection(event) {
+    const selection = document.getSelection();
+    if (selection.rangeCount === 0) {
+      return;
+    }
+    const range = selection.getRangeAt(0);
+    if (range.collapsed) {
+      return;
+    }
+
+    if (!this.#textLayer?.div.contains(range.commonAncestorContainer)) {
+      return;
+    }
+
+    const {
+      x: layerX,
+      y: layerY,
+      width: parentWidth,
+      height: parentHeight,
+    } = this.#textLayer.div.getBoundingClientRect();
+    const bboxes = range.getClientRects();
+
+    // We must rotate the boxes because we want to have them in the non-rotated
+    // page coordinates.
+    let rotator;
+    switch (this.viewport.rotation) {
+      case 90:
+        rotator = (x, y, w, h) => ({
+          x: (y - layerY) / parentHeight,
+          y: 1 - (x + w - layerX) / parentWidth,
+          width: h / parentHeight,
+          height: w / parentWidth,
+        });
+        break;
+      case 180:
+        rotator = (x, y, w, h) => ({
+          x: 1 - (x + w - layerX) / parentWidth,
+          y: 1 - (y + h - layerY) / parentHeight,
+          width: w / parentWidth,
+          height: h / parentHeight,
+        });
+        break;
+      case 270:
+        rotator = (x, y, w, h) => ({
+          x: 1 - (y + h - layerY) / parentHeight,
+          y: (x - layerX) / parentWidth,
+          width: h / parentHeight,
+          height: w / parentWidth,
+        });
+        break;
+      default:
+        rotator = (x, y, w, h) => ({
+          x: (x - layerX) / parentWidth,
+          y: (y - layerY) / parentHeight,
+          width: w / parentWidth,
+          height: h / parentHeight,
+        });
+        break;
+    }
+
+    const boxes = [];
+    for (const { x, y, width, height } of bboxes) {
+      if (width === 0 || height === 0) {
+        continue;
+      }
+      boxes.push(rotator(x, y, width, height));
+    }
+    this.#createAndAddNewEditor(event, false, {
+      boxes,
+    });
```

Comment:
Could the `boxes` Array ever be empty at this point, and if so would it make sense to add a check (with an early return) for that case here?